cmd/terramate/cli/script_tree.go

```diff
@@ -0,0 +1,164 @@
+// Copyright 2023 Terramate GmbH
+// SPDX-License-Identifier: MPL-2.0
+
+package cli
+
+import (
+	"io"
+	"path"
+	"sort"
+	"strings"
+
+	"github.com/fatih/color"
+	"github.com/terramate-io/terramate/config"
+	"github.com/terramate-io/terramate/hcl"
+	project2 "github.com/terramate-io/terramate/project"
+	"golang.org/x/exp/slices"
+)
+
+func (c *cli) printScriptTree() {
+	srcpath := project2.PrjAbsPath(c.rootdir(), c.wd())
+
+	cfg, found := c.cfg().Lookup(srcpath)
+	if !found {
+		return
+	}
+
+	rootNode, topNode := addParentScriptTreeNodes(cfg, nil, true)
+	addChildScriptTreeNodes(cfg, topNode)
+
+	var sb strings.Builder
+	rootNode.format(&sb, "", nil)
+	c.output.MsgStdOut(sb.String())
+}
+
+type scriptsTreeNode struct {
+	DirName   string
+	StackName string
+	Scripts   []*hcl.Script
+	Children  []*scriptsTreeNode
+}
+
+func (node *scriptsTreeNode) format(w io.Writer, prefix string, parentScripts []string) {
+	stackColor := color.New(color.FgGreen).SprintFunc()
+	scriptColor := color.New(color.FgYellow).SprintFunc()
+	parentScriptColor := color.New(color.Faint).SprintFunc()
+
+	write := func(s string) {
+		_, _ = io.WriteString(w, s)
+	}
+
+	writeln := func(s string) {
+		_, _ = io.WriteString(w, s+"\n")
+	}
+
+	var text string
+	if node.DirName != "" {
+		text += node.DirName
+	} else {
+		text += "/"
+	}
+	writeln(text)
+
+	blockPrefix := prefix
+	if len(node.Children) > 0 {
+		blockPrefix += "│ "
+	} else {
+		blockPrefix += "  "
+	}
+
+	if node.StackName != "" {
+		writeln(blockPrefix + "# " + stackColor(node.StackName))
```

Comment:
we could use:
```
fmt.Fprintln(w, blockPrefix + "# " + stackColor(node.StackName))
```

and `fmt.Fprint(w, ...)` later in this file.

Then we could remove the `writeln` + `write` functions. What do you think?

Reply:
Yes, that's nicer, thanks!